### PR TITLE
[TechDebt]: Remove use of `errs.Must` within service packages

### LINKED ISSUE #39038: [TechDebt]: Reduce use of `errs.Must`
### Description

`errs.Must` is overused within the `internal/services` sub-packages, most often wrapping calls to the `flex.FlattenResourceId` function. 

```console
% rg "errs.Must\(flex.FlattenResourceId" --count | wc -l
37
```

These errors should be explicitly handled and returned via a diagnostic rather than causing the provider to panic. 

### References

- Relates https://github.com/hashicorp/terraform-provider-aws/issues/38846

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by jar-b**
### Description

This check will match on any use of `errs.Must(...)` within `internal/service/**/`. Errors should be handled explicitly in service packages to avoid the possibility of crashing the provider during execution. This function was originally intended for use while initializing service clients during the `ConfigureProvider` operation, where a panic is the appropriate course of action.

### Relations

Closes #39038

### Output from Acceptance Testing

```console
% make testacc PKG=bedrockagent TESTS="TestAccBedrockAgent_serial/DataSource/basic|TestAccBedrockAgentAgentActionGroup_basic|TestAccBedrockAgentAgentAlias_basic|TestAccBedrockAgentAgentKnowledgeBaseAssociation_basic"
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/bedrockagent/... -v -count 1 -parallel 20 -run='TestAccBedrockAgent_serial/DataSource/basic|TestAccBedrockAgentAgentActionGroup_basic|TestAccBedrockAgentAgentAlias_basic|TestAccBedrockAgentAgentKnowledgeBaseAssociation_basic'  -timeout 360m

--- PASS: TestAccBedrockAgentAgentAlias_basic (33.15s)
--- PASS: TestAccBedrockAgentAgentActionGroup_basic (34.79s)
--- PASS: TestAccBedrockAgentAgentKnowledgeBaseAssociation_basic (1610.63s)
--- PASS: TestAccBedrockAgent_serial (1664.05s)
    --- PASS: TestAccBedrockAgent_serial/DataSource (1664.05s)
        --- PASS: TestAccBedrockAgent_serial/DataSource/basic (1664.05s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/bedrockagent       1669.423s
```

```console
% make testacc PKG=cloudformation TESTS=TestAccCloudFormationStackInstances_basic
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/cloudformation/... -v -count 1 -parallel 20 -run='TestAccCloudFormationStackInstances_basic'  -timeout 360m

--- PASS: TestAccCloudFormationStackInstances_basic (81.52s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/cloudformation     88.170s
```

```console
% make testacc PKG=cognitoidp TESTS=TestAccCognitoIDPUserPoolUICustomization_Client_image
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/cognitoidp/... -v -count 1 -parallel 20 -run='TestAccCognitoIDPUserPoolUICustomization_Client_image'  -timeout 360m

--- PASS: TestAccCognitoIDPUserPoolUICustomization_Client_image (31.82s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/cognitoidp 37.177s
```

```console
% make testacc PKG=cloudfrontkeyvaluestore TESTS=TestAccCloudFrontKeyValueStoreKey_basic
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/cloudfrontkeyvaluestore/... -v -count 1 -parallel 20 -run='TestAccCloudFrontKeyValueStoreKey_basic'  -timeout 360m

--- PASS: TestAccCloudFrontKeyValueStoreKey_basic (110.98s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/cloudfrontkeyvaluestore    117.716s
```

```console
% make testacc PKG=connect TESTS=TestAccConnect_serial/LambdaFunctionAssociation/basic
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/connect/... -v -count 1 -parallel 20 -run='TestAccConnect_serial/LambdaFunctionAssociation/basic'  -timeout 360m

--- PASS: TestAccConnect_serial (175.23s)
    --- PASS: TestAccConnect_serial/LambdaFunctionAssociation (175.23s)
        --- PASS: TestAccConnect_serial/LambdaFunctionAssociation/basic (89.25s)
        --- PASS: TestAccConnect_serial/LambdaFunctionAssociation/dataSource_basic (85.98s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/connect    181.961s
```

```console
% make testacc PKG=ebs TESTS=TestAccEC2EBSFastSnapshotRestore_basic
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/ec2/... -v -count 1 -parallel 20 -run='TestAccEC2EBSFastSnapshotRestore_basic'  -timeout 360m

--- PASS: TestAccEC2EBSFastSnapshotRestore_basic (168.27s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/ec2        175.209s
```

```console
% make testacc PKG=elbv2 TESTS=TestAccELBV2TrustStoreRevocation_basic
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/elbv2/... -v -count 1 -parallel 20 -run='TestAccELBV2TrustStoreRevocation_basic'  -timeout 360m

--- PASS: TestAccELBV2TrustStoreRevocation_basic (43.79s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/elbv2      49.271s
```

```console
% make testacc PKG=elasticache TESTS=TestAccElastiCacheUserGroupAssociation_basic
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/elasticache/... -v -count 1 -parallel 20 -run='TestAccElastiCacheUserGroupAssociation_basic'  -timeout 360m
=== RUN   TestAccElastiCacheUserGroupAssociation_basic
=== PAUSE TestAccElastiCacheUserGroupAssociation_basic
=== CONT  TestAccElastiCacheUserGroupAssociation_basic
--- PASS: TestAccElastiCacheUserGroupAssociation_basic (257.08s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/elasticache        262.428s
```

```console
% make testacc PKG=lambda TESTS="TestAccLambdaLayerVersionPermission_basic_byARN|TestAccLambdaLayerVersionPermission_basic_byName|TestAccLambdaProvisionedConcurrencyConfig_basic"
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/lambda/... -v -count 1 -parallel 20 -run='TestAccLambdaLayerVersionPermission_basic_byARN|TestAccLambdaLayerVersionPermission_basic_byName|TestAccLambdaProvisionedConcurrencyConfig_basic'  -timeout 360m

--- PASS: TestAccLambdaLayerVersionPermission_basic_byName (15.52s)
--- PASS: TestAccLambdaLayerVersionPermission_basic_byARN (19.29s)
--- PASS: TestAccLambdaProvisionedConcurrencyConfig_basic (134.42s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/lambda     139.738s
```

```console
% make testacc PKG=m2 TESTS=TestAccM2Deployment_basic
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/m2/... -v -count 1 -parallel 20 -run='TestAccM2Deployment_basic'  -timeout 360m

--- PASS: TestAccM2Deployment_basic (2507.94s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/m2 2514.812s
```

```console
% make testacc PKG=networkmonitor TESTS=TestAccNetworkMonitorProbe_basic
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/networkmonitor/... -v -count 1 -parallel 20 -run='TestAccNetworkMonitorProbe_basic'  -timeout 360m
=== RUN   TestAccNetworkMonitorProbe_basic
=== PAUSE TestAccNetworkMonitorProbe_basic
=== CONT  TestAccNetworkMonitorProbe_basic
    probe_test.go:34: Step 3/3 error: Error running apply: exit status 1

        Error: waiting for CloudWatch Network Monitor Probe (tf-acc-test-7273269475067989614,probe-5ujg03zhv8segr8lda074dezf) delete

        timeout while waiting for resource to be gone (last state: 'DELETING',
        timeout: 15m0s)
    panic.go:626: Error running post-test destroy, there may be dangling resources: exit status 1

        Error: deleting CloudWatch Network Monitor Probe (tf-acc-test-7273269475067989614,probe-5ujg03zhv8segr8lda074dezf)

        operation error NetworkMonitor: DeleteProbe, exceeded maximum number of
        attempts, 25, https response error StatusCode: 500, RequestID:
        843deb04-f53d-46e6-b3d1-583dae3cf4c8, InternalServerException: Unexpected
        error occurred.
--- FAIL: TestAccNetworkMonitorProbe_basic (4410.10s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-aws/internal/service/networkmonitor     4416.932s
```

☝️ Error is pre-existing and occurring in CI.

```console
% make testacc PKG=route53 TESTS="TestAccRoute53CIDRLocation_basic|TestAccRoute53KeySigningKey_basic"
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/route53/... -v -count 1 -parallel 20 -run='TestAccRoute53CIDRLocation_basic|TestAccRoute53KeySigningKey_basic'  -timeout 360m

--- PASS: TestAccRoute53CIDRLocation_basic (11.21s)
--- PASS: TestAccRoute53KeySigningKey_basic (205.49s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/route53    210.870s
```

```console
% make testacc PKG=s3control TESTS="TestAccS3ControlAccessGrants_serial/Location/basic|TestAccS3ControlAccessGrants_serial/Grant/basic"
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/s3control/... -v -count 1 -parallel 20 -run='TestAccS3ControlAccessGrants_serial/Location/basic|TestAccS3ControlAccessGrants_serial/Grant/basic'  -timeout 360m

--- PASS: TestAccS3ControlAccessGrants_serial (48.42s)
    --- PASS: TestAccS3ControlAccessGrants_serial/Location (24.39s)
        --- PASS: TestAccS3ControlAccessGrants_serial/Location/basic (24.39s)
    --- PASS: TestAccS3ControlAccessGrants_serial/Grant (24.03s)
        --- PASS: TestAccS3ControlAccessGrants_serial/Grant/basic (24.03s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/s3control  55.084s
```

```console
% make testacc PKG=ssm TESTS=TestAccSSMPatchGroup_basic
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/ssm/... -v -count 1 -parallel 20 -run='TestAccSSMPatchGroup_basic'  -timeout 360m

--- PASS: TestAccSSMPatchGroup_basic (8.71s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/ssm        14.115s
```

```console
% make testacc PKG=wafv2 TESTS=TestAccWAFV2WebACLAssociation_basic
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.7 test ./internal/service/wafv2/... -v -count 1 -parallel 20 -run='TestAccWAFV2WebACLAssociation_basic'  -timeout 360m

--- PASS: TestAccWAFV2WebACLAssociation_basic (46.10s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/wafv2      51.404s
```

### CODE REVIEW

File: internal/service/account/region.go [hunk 1]
```diff
@@ -78,7 +77,10 @@ func resourceRegionUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 	accountID := ""
 	if v, ok := d.GetOk(names.AttrAccountID); ok {
 		accountID = v.(string)
-		id = errs.Must(flex.FlattenResourceId([]string{accountID, region}, regionResourceIDPartCount, false))
+		id, err := flex.FlattenResourceId([]string{accountID, region}, regionResourceIDPartCount, false)
+		if err != nil {
+			return sdkdiag.AppendErrorf(diags, "enabling Account Region (%s): %s", id, err)
```
Comment (ewbankkit) [inline internal/service/account/region.go:82]: I guess `id` will always be `""` here (and in other resources) as `err != nil`.
May be simpler to just `return sdkdiag.AppendFromErr(diags, err)`? These cases occur rarely and Terraform core may provide enough context (operation and resource name).

Comment (jar-b) [reply internal/service/account/region.go]: Fixed!

File: internal/service/bedrockagent/agent_alias.go [hunk 2]
```diff
@@ -137,7 +137,12 @@ func (r *agentAliasResource) Create(ctx context.Context, request resource.Create
 
 	// Set values for unknowns.
 	data.AgentAliasID = fwflex.StringToFramework(ctx, output.AgentAlias.AgentAliasId)
-	data.setID()
+	id, err := data.setID()
+	if err != nil {
+		response.Diagnostics.AddError("flattning resource ID Bedrock Agent Alias", err.Error())
```
Comment (ewbankkit) [inline internal/service/bedrockagent/agent_alias.go:142]:
```suggestion
		response.Diagnostics.AddError("creating Bedrock Agent Alias", err.Error())
```

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

**Pull request merged**